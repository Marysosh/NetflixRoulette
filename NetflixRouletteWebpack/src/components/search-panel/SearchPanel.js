import React from "react";
import "./SearchPanel.scss";

import SearchForm from "../search-form/SearchForm";

function SearchPanel() {
  return (
    <div className="search-panel">
      <div className="search-panel__search-form">
        <SearchForm />
      </div>
    </div>
  );
}

export default SearchPanel;
