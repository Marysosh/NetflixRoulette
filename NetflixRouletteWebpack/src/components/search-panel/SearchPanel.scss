.search-panel {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  background: url("SearchPanelImg.png") no-repeat;
  background-size: 100% 100%;
  height: 56vh;

  &__title,
  &__search-form {
    margin: 0 20vh;
  }

  &__title {
    margin-top: 2vh;
  }

  &__search-form {
    display: flex;
    height: 8vh;
    margin-bottom: 20vh;
  }
}
