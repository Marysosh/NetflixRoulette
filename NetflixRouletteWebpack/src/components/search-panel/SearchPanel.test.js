import React from "react";
import { Provider } from "react-redux";
import { BrowserRouter as Router } from "react-router-dom";
import { render } from "@testing-library/react";
import configureStore from "redux-mock-store";

import SearchPanel from "./SearchPanel";

const mockStore = configureStore([]);

describe("Search Panel component", () => {
  let store;

  beforeEach(() => {
    store = mockStore({
      modalsManagement: {
        isAddMovieModalOpen: false,
        isCongratsModalOpen: false,
      },
      movieList: {
        selectedFilters: [],
        sortingType: "release_date",
        sortingOrder: "desc",
      },
    });
  });

  it("renders correctly", () => {
    const { asFragment } = render(
      <Provider store={store}>
        <Router>
          <SearchPanel />
        </Router>
      </Provider>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
