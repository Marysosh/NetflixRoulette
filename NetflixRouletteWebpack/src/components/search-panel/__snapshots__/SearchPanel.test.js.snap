// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Panel component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="search-panel"
  >
    <div
      class="header"
    >
      <div
        class="logo"
      >
        <img
          alt="logo"
          src="test-file-stub"
        />
      </div>
      <button
        class="add-movie-btn"
        type="button"
      >
        + Add movie
      </button>
    </div>
    <div
      class="search-panel__title"
    >
      <h2
        class="search-title"
      >
        Find your movie
      </h2>
    </div>
    <div
      class="search-panel__search-form"
    >
      <form>
        <div
          class="search-form"
        >
          <input
            class="search-input"
            placeholder="What do you want to watch?"
            type="text"
          />
          <button
            class=" filled-btn"
            type="button"
          >
            Search
          </button>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
